tests/zfs-tests/include/libtest.shlib

```diff
@@ -3026,7 +3041,7 @@ function vdevs_in_pool
 	shift
 
 	typeset tmpfile=$(mktemp)
-	zpool list -Hv "$pool" >$tmpfile
+	zpool status -v "$pool" | grep -A 1000 "config:" >$tmpfile
```

Comment:
Thanks for the detailed answer. That makes sense. Maybe putting  a quick comment right above the zpool status could be helpful - saying that we could use `zpool list` and only get the vdevs of that pool but we can't reference a mirror as a whole by its ID, therefore we use `zpool status`. 

An alternative idea would be to change the new test and `remove_mirror_sanity.ksh` (I think that's the only other one that tries the same thing) and make both of them pass the 2 child vdevs instead of the name mirror-X. (And maybe put a note in the comment of the current function that consumers should be careful and use the leaf vdevs in this function for raidz, mirrors, etc..).

Reply:
Comment added.  I stuck with the 'zpool status' solution since it seems entirely reasonable to me that should be supported.